instrumentation/rxjava-2.0/library/src/main/java/io/opentelemetry/instrumentation/rxjava2/TracingAssembly.java

```diff
@@ -0,0 +1,318 @@
+import io.reactivex.CompletableObserver;
+import io.reactivex.Flowable;
+import io.reactivex.Maybe;
+import io.reactivex.MaybeObserver;
+import io.reactivex.Observable;
+import io.reactivex.Observer;
+import io.reactivex.Single;
+import io.reactivex.SingleObserver;
+import io.reactivex.annotations.NonNull;
+import io.reactivex.flowables.ConnectableFlowable;
+import io.reactivex.functions.BiFunction;
+import io.reactivex.functions.Function;
+import io.reactivex.internal.fuseable.ConditionalSubscriber;
+import io.reactivex.observables.ConnectableObservable;
+import io.reactivex.parallel.ParallelFlowable;
+import io.reactivex.plugins.RxJavaPlugins;
+import org.checkerframework.checker.lock.qual.GuardedBy;
+import org.reactivestreams.Subscriber;
+
+public class TracingAssembly {
+
+  @SuppressWarnings("rawtypes")
+  @GuardedBy("TracingAssembly.class")
+  private static Function<? super Observable, ? extends Observable> oldOnObservableAssembly;
+
+  @SuppressWarnings("rawtypes")
+  @GuardedBy("TracingAssembly.class")
+  private static BiFunction<? super Observable, ? super Observer, ? extends Observer>
+      oldOnObservableSubscribe;
+
+  @SuppressWarnings("rawtypes")
+  @GuardedBy("TracingAssembly.class")
+  private static Function<? super ConnectableObservable, ? extends ConnectableObservable>
+      oldOnConnectableObservableAssembly;
+
+  @SuppressWarnings("rawtypes")
+  @GuardedBy("TracingAssembly.class")
+  private static BiFunction<
+          ? super Completable, ? super CompletableObserver, ? extends CompletableObserver>
+      oldOnCompletableSubscribe;
+
+  @SuppressWarnings("rawtypes")
+  @GuardedBy("TracingAssembly.class")
+  private static BiFunction<? super Single, ? super SingleObserver, ? extends SingleObserver>
+      oldOnSingleSubscribe;
+
+  @SuppressWarnings("rawtypes")
+  @GuardedBy("TracingAssembly.class")
+  private static BiFunction<? super Maybe, ? super MaybeObserver, ? extends MaybeObserver>
+      oldOnMaybeSubscribe;
+
+  @SuppressWarnings("rawtypes")
+  @GuardedBy("TracingAssembly.class")
+  private static BiFunction<? super Flowable, ? super Subscriber, ? extends Subscriber>
+      oldOnFlowableSubscribe;
+
+  @SuppressWarnings("rawtypes")
+  @GuardedBy("TracingAssembly.class")
+  private static Function<? super ConnectableFlowable, ? extends ConnectableFlowable>
+      oldOnConnectableFlowableAssembly;
+
+  @SuppressWarnings("rawtypes")
+  @GuardedBy("TracingAssembly.class")
+  private static Function<? super ParallelFlowable, ? extends ParallelFlowable>
+      oldOnParallelAssembly;
+
+  @GuardedBy("TracingAssembly.class")
+  private static boolean enabled;
+
+  private TracingAssembly() {}
+
+  @SuppressWarnings({"rawtypes", "unchecked"})
+  public static synchronized void enable() {
+    if (enabled) {
+      return;
+    }
+
+    enableObservable();
+
+    oldOnConnectableObservableAssembly = RxJavaPlugins.getOnConnectableObservableAssembly();
+    RxJavaPlugins.setOnConnectableObservableAssembly(
```

Comment:
Armeria's `RequestContextAssembly` and this class use a slightly different set of rxjava plugins (and also `RxJavaPlugins` has several other ones) - do both classes cover the same functionality? Can you add a javadoc comment in this class explaining the choice of functions to instrument?